src/inventory_ui.cpp

```diff
@@ -2908,3 +2917,79 @@ inventory_selector::stats inventory_drop_selector::get_raw_stats() const
                u.volume_capacity_with_tweaks( to_use ),
                u.max_single_item_length(), u.max_single_item_volume() );
 }
+
+void inventory_examiner::action_examine( const item_location sitem, const bool fit_to_window )
+{
+    if( !fit_to_window ) {
+        inventory_selector::action_examine( sitem );
+    } else {
+        // Code below pulled from the action_examine function in advanced_inv.cpp
+        std::vector<iteminfo> vThisItem;
+        std::vector<iteminfo> vDummy;
+
+        sitem->info( true, vThisItem );
+
+        item_info_data data( sitem->tname(), sitem->type_name(), vThisItem, vDummy, examine_window_scroll );
+        //Tries to create a window covering the right half of the inventory screen
+        data.without_getch = true;
+
+        // NOTE: This assumes that the first column (currently own_inv_column) is used for the item list:
+        int inv_column_width = get_all_columns().front()->get_width();
+        /* NOTE: By flagging force_max_window_size as true, this forces the inventory_selector window to be
+         * TERMX x TERMY , allowing us to assume the window starts at 0,0 and is TERMX by TERMY.  If this is
+         * changed in inventory_selector::prepare_layout, then everything breaks.  This obviously isn't a
+         * great solution */
+
+        int border_width = 1; //Assumed, since the value in inventory_selector is private, but reasonable
+        int width = TERMX - 2 * border_width  - inv_column_width;
+        int height = TERMY  - get_header_height() - 1;
+        point start_position = point( ( inv_column_width + border_width + 1 ),
+                                      get_header_height() + 1 );
+
+        draw_item_info( catacurses::newwin( height, width, start_position ), data );
+    }
+}
+
+item_location inventory_examiner::execute()
+{
+    shared_ptr_fast<ui_adaptor> ui = create_or_get_ui_adaptor();
+    while( true ) {
+        ui_manager::redraw();
+
+        const inventory_entry &selected = get_active_column().get_selected();
+        if( selected ) {
+            const item_location &sitem = selected.any_item();
+            action_examine( sitem, true );
+        }
+
+        const inventory_input input = get_input();
+
+        if( input.entry != nullptr ) {
+            if( select( input.entry->any_item() ) ) {
+                ui_manager::redraw();
+            }
+            return input.entry->any_item();
+        }
+
+        const int scroll_item_info_lines = get_layout_height() - 4;
+
+        if( input.action == "QUIT" ) {
+            return item_location();
+        } else if( input.action == "PAGE_UP" ) {
+            examine_window_scroll -= scroll_item_info_lines;
+        } else if( input.action == "PAGE_DOWN" ) {
+            examine_window_scroll += scroll_item_info_lines;
+        } else if( input.action == "CONFIRM" ) {
+            const inventory_entry &selected = get_active_column().get_selected();
```

Comment:
You've already declared this `selected` above and it shouldn't have changed in the meantime.

Reply:
Agreed.  That line can be removed.